File: packages/ios/Sources/CivUI/CivReadMore.swift
// CivUI — CivReadMore for SwiftUI
// Two-stage content disclosure (mirrors the web civ-read-more component).
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire a VStack with the teaser always visible
// and the rest region show/hidden by an internal `@State` boolean.
// See audit-debt.md for status.

import SwiftUI

/// Accessible two-stage disclosure: a teaser paragraph stays visible
/// at all times; a "Read more" button reveals additional content.
public struct CivReadMore: View {
    /// Whether the rest region is currently visible.
    @Binding public var open: Bool

    /// Override the "Read more" trigger text.
    public var moreLabel: String

    /// Override the "Read less" trigger text.
    public var lessLabel: String

    /// Optional icon name shown before the label.
    public var icon: String

    /// Trigger size: "default" or "sm". DEPRECATED — use `spacing` instead.
    public var size: String

    /// Trigger padding density: "default" or "sm".
    public var spacing: String

    /// Render inline so the trigger flows as the last words of the teaser.
    public var isInline: Bool

    /// Opt out of the default fade-and-overlay treatment (block mode only).
    /// When true, the trigger sits below the teaser as a plain button instead
    /// of floating over a gradient at the bottom of the text.
    public var isNoFadeTrigger: Bool

    /// Called when the open state changes.
    public var onToggle: ((Bool) -> Void)?

    /// Called for analytics tracking.
    public var onAnalytics: ((String, [String: Any]?) -> Void)?

    public init(
        open: Binding<Bool> = .constant(false),
        moreLabel: String = "",
        lessLabel: String = "",
        icon: String = "",
        size: String = "default",
        spacing: String = "default",
        isInline: Bool = false,
        isNoFadeTrigger: Bool = false,
        onToggle: ((Bool) -> Void)? = nil,
        onAnalytics: ((String, [String: Any]?) -> Void)? = nil
    ) {
        self._expanded = open
        self.moreLabel = moreLabel
        self.lessLabel = lessLabel
        self.icon = icon
        self.size = size
        self.spacing = spacing
        self.isInline = isInline
        self.isNoFadeTrigger = isNoFadeTrigger
        self.onToggle = onToggle
        self.onAnalytics = onAnalytics
    }

    public var body: some View {
        EmptyView()
    }
}

File: packages/android/src/main/kotlin/gov/civui/components/CivReadMore.kt
// CivUI — CivReadMore for Jetpack Compose
// Two-stage content disclosure (mirrors the web civ-read-more component).
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire a Compose Column with the teaser always
// visible and an AnimatedVisibility for the rest region. See
// audit-debt.md for status.

package gov.civui.components

import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * Accessible two-stage disclosure: a teaser paragraph stays visible
 * at all times; a "Read more" button reveals additional content.
 */
@Composable
fun CivReadMore(
    open: Boolean = false,
    moreLabel: String = "",
    lessLabel: String = "",
    icon: String = "",
    /**
     * Trigger padding density: "default" or "sm". Matches the
     * design-system `spacing` convention used by `CivAlert`, `CivCard`,
     * `CivListItem`, and the data-grid input controls.
     */
    spacing: String = "default",
    /**
     * Trigger size: "default" or "sm". **Deprecated** — use `spacing`
     * instead. Same allowed values, same effect. Will be removed in a
     * future release.
     */
    size: String = "default",
    inline: Boolean = false,
    noFadeTrigger: Boolean = false,
    modifier: Modifier = Modifier,
    onToggle: ((Boolean) -> Unit)? = null,
    onAnalytics: ((event: String, data: Map<String, Any>?) -> Unit)? = null,
    teaser: @Composable () -> Unit = {},
    rest: @Composable () -> Unit = {},
) {
    Column(modifier = modifier) {
        // Placeholder — see audit-debt.md for native implementation status.
    }
}
